[SWE-20] Add ESLint & Husky

jackfales opened the pull request:
Added ESLint and Husky pre-commit hook to run the linter before each commit. Added `lint-staged` to run the linter only on staged files. Updated README with commands to run the linter manually with or without fixes.

datderek commented on .husky/pre-commit:
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+npm run lint:fix
```
```suggestion
npx lint-staged
```
https://github.com/okonet/lint-staged#examples

datderek commented on package.json:
```diff
@@ -3,6 +3,8 @@
     "dev": "next dev",
+    "lint": "next lint",
+    "lint:fix": "next lint --fix",
     "start": "next start"
   },
```
```suggestion
    "start": "next start"
    "prepare": "husky install"
  },
```
Do we need a "prepare" script to enable the Git hooks? Pre-commit didn't run for me otherwise.

datderek reviewed (changes requested):
Some issue running the Git hook, may need a prepare script to run `husky install` on `npm i`. Update precommit hook to only run on staged files. Otherwise LGTM.

jackfales commented:
> Husky precommit hook didn't run, may need a prepare script to run `husky install` on `npm i`. Update precommit hook to only run on staged files. Otherwise LGTM.

Good catch. Updated pre-commit hook and added prepare script.

datderek reviewed (changes requested):
`npx lint-staged` errors without configuration file.

https://nextjs.org/docs/pages/building-your-application/configuring/eslint#lint-staged

jackfales commented:
> `npx lint-staged` errors without configuration file.
> 
> https://nextjs.org/docs/pages/building-your-application/configuring/eslint#lint-staged

Can you add more info? What's the error? What caused it? Is it reproducible? Screenshots help.

jackfales commented:
Changed `lint-staged` to use `eslint` instead of `next lint`. The latter runs into some issues when linting specific files, which is what `lint-staged` does.

datderek reviewed (approved):
Looks good to me!

sammyoto reviewed (approved):
linter works. looks good to me
